### DISCUSSION TITLE
Performance of different semirings

### DISCUSSION THREAD

**Discussion by SultanOrazbayev** [python-graphblas/python-graphblas#460] [General]
When running specific calculations/algorithms, different semirings can yield the same answer. Is there any guideline/heuristic on which semirings might have better performance (memory footprint or speed)?

**Comment by eriknw** [comment]
Great question. Do you have specific examples to compare?

In SuiteSpare:GraphBLAS, it's generally recommended to avoid "GENERIC" operations, which can occur for many reasons. To see whether an operation is "GENERIC", enable the "burble" and look for "generic" in the printed diagnostic info. For example:
```python
gb.ss.config["burble"] = True
```
However, generic operations shouldn't be an issue once the JIT comes in SuiteSparse:GraphBLAS 8.0 (coming soon!). Also, UDFs and UDTs are generally slower too.

Many of the operations used in `graphblas-algorithms` were compared to alternatives and chosen b/c they are generally faster.

I don't know how to answer this question in general other than "benchmark and compare". It may be nice to gether recommendations here for specific operations.

**Reply by SultanOrazbayev** [reply]
Thank you, Erik! It would help if I had a specific example, but this was just a general thought I had when going through my notes.
